src/core/net/srp_server.cpp

```diff
@@ -718,6 +718,8 @@ void Server::ProcessDnsUpdate(Message &aMessage, MessageMetadata &aMetadata)
     // Parse lease time and validate signature.
     SuccessOrExit(error = ProcessAdditionalSection(host, aMessage, aMetadata));
 
+    SuccessOrExit(error = ValidateServiceSubTypes(*host, aMetadata));
+
```

Comment:
I feel it may be better doing this check after parsing the entire SRP update message (and additional section where lease the lease times). So we know everything from message (for example if msg is being removed or not - though I think we may not  currently use this during the check but I feel safer having it everything parsed).

Reply:
Sure, not a big issue to me~